services/src/main/groovy/org/jfrog/artifactory/client/ArtifactoryClientBuilder.java

```diff
@@ -112,7 +120,12 @@ private CloseableHttpClient createClientBuilder(URI uri) {
             artifactoryHttpClient.socketTimeout(socketTimeout);
         }
 
-        artifactoryHttpClient.trustSelfSignCert(!ignoreSSLIssues);
+        if (sslContextBuilder != null) {
+            artifactoryHttpClient.sslContextBuilder(sslContextBuilder);
+        }
+        else {
```

Comment:
What is the reason for this "else"?

Reply:
There's a nasty bug with this `artifactoryHttpClient.trustSelfSignCert(!ignoreSSLIssues)`.
By default, `ignoreSSLIssues` is `false`, and `trustSelfSignCert` is therefore `true`. And when I set `ignoreSSLIssues` to `true`, I end up with self-signed certs not being trusted (!)

Without this `if/else`, because I don't want `trustSelfSignCert` to be `true`, I'll have to set `ignoreSSLIssues`, which is rather awkward.

Do you want me to fix this `ignoreSSLIssues` issue at the same time?